docs/developer-docs/latest/setup-deployment-guides/typescript.md

```diff
@@ -81,9 +81,39 @@ yarn develop
 
 </code-group>
 
-<!--## Use TypeScript typings
+## Use TypeScript typings
+
+Strapi provides typings on the Strapi class to improve the TypeScript experience. In the following example a `register` method is added to `./src/index.ts`. For example, when the `strapi.runLifecyclesFunctions` method is added, a list of available lifecycle types (`register`, `bootstrap` and `destroy`) are returned by the code editor. The following example is illustrated statically and the second tab shows an animated example.
```

Comment:
_(Roughly copying, pasting, and adapting a bit what I already mentioned on Slack, for the record :-) )_

After second thought, I don't think the GIF is a big issue after all (except for accessibility issues, see below). Maybe we could try to slow it down a bit (if doable) since it can't be paused like a video would.

However, with my limited understanding of TypeScript so far 😅  I'm just wondering:
Are this GIF/static example showing typings or rather autocomplete?

I guess the IDE shows the available methods because we specified that `strapi` is an instance of the Strapi class — i.e. the `strapi` parameter is of type `Strapi` (`strapi: Strapi`), right? (and this is where TypeScript typings matter). We could probably go a bit deeper into the explanations 🤔 
If @Convly or some dev could have some time to explain, I'd love to read/hear more info about the topic 🙂 

However, my suggestion would be that, if we are indeed focusing more on autocomplete around typings, it would probably be best to convert the "static example" to a procedure. This way we describe what people will really experience. This also alleviates accessibility issues (people using screen readers will still have the GIF somewhat described by the picture)

Something like:
1. In the register method, declare the strapi argument as of type Strapi
2. Type strapi. and use keyboard arrows to browse the available interfaces.
3. Choose runLifecyclesfunctions  from the list.
4. Within the function arguments, choose one of the available functions (e.g. register, boostrap…)
Missing some steps and needs to be proofread/copyedited but you get the idea 🙂 

What do you think?